File: Java/CleanString.java
import java.util.*;
import java.lang.*;

public class CleanString {

  public static void main(String[] args) {
    Scanner scan = new Scanner(System.in);
    boolean shouldRepeat = true;

    do {
      System.out.println();
      System.out.println("Enter a string:");
      String input = scan.nextLine();
      String cleanString = stringClean(input);
      System.out.println(input + " -> " + cleanString);

      System.out.println("Want to go again? (y/n)");
      String choice = scan.nextLine();
      shouldRepeat = choice.equalsIgnoreCase("y");
    } while (shouldRepeat);

    System.out.println("</---------------------------------------------------------------------->");
    scan.close();
  }

  public static String stringClean(String input) {
    String output = "";
    for (int i = 0; i < input.length(); ++i) {
      CharSequence c = String.valueOf(input.charAt(i));
      if (!output.contains(c)) {
        output += c;
      }
    }
    return output;
  }

}

File: Kotlin/CleanString.kt
package CleanString

import java.lang.*
import java.util.*

fun main(args: Array<String>) {
  val scan = Scanner(System.`in`)
  var shouldRepeat: Boolean
  do {
    println()
    println("Enter a string:")
    var input = scan.nextLine()
    val cleanString = stringClean(input)
    println("$input -> $cleanString")

    println("Want to go again? (y/n)")
    var choice = scan.nextLine()
    shouldRepeat = choice.equals("y", true)
  } while (shouldRepeat)
  println("</---------------------------------------------------------------------->")
}

fun stringClean(input: String): String {
  var output = ""
  input.forEach { if (!output.contains(it)) output += it }
  return output
}
